lib/compilers/d8.ts

```diff
@@ -79,7 +79,7 @@ export class D8Compiler extends BaseCompiler {
         let outputFilename = '';
 
         // Instantiate Java or Kotlin compiler based on the current language.
```

Comment:
If you're changing the code, please update the comment too. Though: why was this change needed? I don't really like this hack but is there another way to do this?

Reply:
I had mentioned this in the commit message but had forgotten to update the comment here, sorry!

I've actually reverted the change here. I originally updated this because there seemed to be a bug where the underlying D8Compiler defaulted to android-java, so compilation when selecting Android Java would work, but not Android Kotlin. I used the source file extension to get around this. However, with the update to add the language name to the Android compilers' IDs, this isn't an issue any more.

As a side note, looking at this also led me to find that I had left in a mistake by using 'android-d8-default' as the language name—I didn't notice this during local testing, so this might have been an error I made later on.

Thanks!